ProcessLib/HydroMechanics/CreateHydroMechanicsProcess.cpp

```diff
@@ -189,7 +194,8 @@ std::unique_ptr<Process> createHydroMechanicsProcess(
         biot_coefficient,
         porosity,
         solid_density,
-        specific_body_force};
+        specific_body_force,
+        *reference_temperature};
```

Comment:
I think this is undefined behaviour in case the config parameter is empty.
https://en.cppreference.com/w/cpp/utility/optional/operator*

Reply:
Do you mean to change the type of the argument to std::optional?

Reply:
or something like
```
(reference_temperature) ? *reference_temperature : [a default value]
```